### ISSUE TITLE
Lint nonminimal-bool: suggests current code

### ISSUE THREAD

**Issue by mvuets** [rust-lang/rust-clippy#4890]
I've run into a boolean expression that Clippy marks as **clippy::nonminimal-bool**, however the suggested rewrite is exactly the same as the current code. Seems like a minor false positive bug to me.

Minimal reproducible example:
```rust
fn main() {
    let (d1, d2, d3) = (2, 4, 4);
    let _f = d1 == d2 && d2 != d3 || d1 != d2 && d2 == d3;
}
```

Clippy output:
```text
$ cargo clippy -- -D clippy::nonminimal-bool
    Checking nonminbool v0.1.0 (/home/mvuets/tmp/nonminbool)
error: this boolean expression can be simplified
 --> src/main.rs:3:14
  |
3 |     let _f = d1 == d2 && d2 != d3 || d1 != d2 && d2 == d3;
  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `d1 == d2 && d2 != d3 || d1 != d2 && d2 == d3`
  |
  = note: requested on the command line with `-D clippy::nonminimal-bool`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#nonminimal_bool
# -- snip --
```

FWIW, the [lint description](https://rust-lang.github.io/rust-clippy/master/index.html#nonminimal_bool) mentions "ignores short circuiting behavior" as a known problem, however I believe that's not the case here.

clippy --version: clippy 0.0.212 (4e7e71b 2019-10-11)

**Comment by mikerite** [comment]
Is this equivalent to `let _f = (d1 == d2) ^ (d2 == d3)`? :thinking:

**Comment by flip1995** [comment]
cc #4548

and @mikerite fixed this already in #4568. At least it doesn't trigger on the [Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2015&gist=883cffa077cbf9e40aaf956d9bec3b6a)
